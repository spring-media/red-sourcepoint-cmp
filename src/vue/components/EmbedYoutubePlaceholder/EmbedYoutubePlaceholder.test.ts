import Vue from 'vue';
import { mount } from '@vue/test-utils';
import { EmbedYoutubePlaceholder } from './';

const loadPrivacyManagerModal = jest.fn();

const privacyManagerStub = Vue.extend({
  name: 'PrivacyManager',
  render() {
    return (
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      this.$scopedSlots.default!({
        loadPrivacyManagerModal,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
      }) as any
    );
  },
});

describe('EmbedYoutubePlaceholder', () => {
  afterEach(() => {
    loadPrivacyManagerModal.mockReset();
  });

  it('should render without any errors', () => {
    expect(mount(EmbedYoutubePlaceholder, { propsData: { privacyManager: 12345 } }).element).toMatchSnapshot();
  });

  describe('should open a privacy manager by clicking on', () => {
    it.each([
      ['the "activate consent" button', '.embed-placeholder__button'],
      ['the link to the description', '.embed-placeholder__link-description'],
      ['the link to the vendor list', '.embed-placeholder__link-vendor-list'],
    ])('%s', (_, selector) => {
      const wrapper = mount(EmbedYoutubePlaceholder, {
        stubs: { PrivacyManager: privacyManagerStub },
        propsData: { privacyManagerId: 12345 },
      });

      wrapper.find(selector).trigger('click');

      expect(loadPrivacyManagerModal).toHaveBeenCalledWith(12345);
      expect(loadPrivacyManagerModal).toHaveBeenCalledTimes(1);
    });
  });
});
