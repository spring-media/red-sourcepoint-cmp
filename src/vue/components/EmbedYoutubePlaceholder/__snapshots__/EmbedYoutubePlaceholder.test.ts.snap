// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedYoutubePlaceholder should render without any errors 1`] = `
<div
  class="embed-placeholder__container"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="172"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <svg
        x="calc(50% - 25px)"
        y="calc(50% - 20px)"
      >
        <g
          transform="scale(0.42)"
        >
          <path
            d="M 118.9,13.3 C 117.5,8.1 113.4,4 108.2,2.6 98.7,0 60.7,0 60.7,0 60.7,0 22.7,0 13.2,2.5 8.1,3.9 3.9,8.1 2.5,13.3 0,22.8 0,42.5 0,42.5 0,42.5 0,62.3 2.5,71.7 3.9,76.9 8,81 13.2,82.4 22.8,85 60.7,85 60.7,85 c 0,0 38,0 47.5,-2.5 5.2,-1.4 9.3,-5.5 10.7,-10.7 2.5,-9.5 2.5,-29.2 2.5,-29.2 0,0 0.1,-19.8 -2.5,-29.3 z"
            fill="#E9ECEF"
          />
           
          <polygon
            fill="#ffffff"
            points="80.2,42.5 48.6,24.3 48.6,60.7"
          />
        </g>
      </svg>
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
    
    An dieser Stelle findest du Inhalte aus YouTube
  
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
    Um mit Inhalten aus YouTube und anderen Sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen wir
    deine Zustimmung.
  
  </div>
   
  <button
    class="embed-placeholder__button"
  >
    
          Soziale Netzwerke aktivieren
        
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir externe Inhalte aus Sozialen Netzwerken angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Mehr dazu findest du in der
        
    <a
      class="embed-placeholder__text-link embed-placeholder__link-description"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Beschreibung dieses Datenverarbeitungszweck
    </a>
    
        und in den Datenschutzinformationen dieser
        
    <a
      class="embed-placeholder__text-link embed-placeholder__link-vendor-list"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Drittanbieter
    </a>
    .
      
  </div>
</div>
`;
